helpdesk_mgmt/controllers/main.py

```diff
@@ -12,8 +14,53 @@ def support_ticket_close(self, **kw):
         values = {}
         for field_name, field_value in kw.items():
             values[field_name] = field_value
-        ticket = http.request.env['helpdesk.ticket'].sudo().\
-            search([('id', '=', values['ticket_id'])])
-        ticket.stage_id = values.get('stage_id')
-
+        ticket = http.request.env['helpdesk.ticket'].sudo().browse([(
+            'id', '=', values['ticket_id'])])
+        stage_id = int(values.get('stage_id'))
+        ticket.sudo().update(
+            {'stage_id': stage_id})
         return werkzeug.utils.redirect("/my/ticket/" + str(ticket.id))
+
+    @http.route('/new/ticket', type="http", auth="user", website=True)
+    def create_new_ticket(self, **kw):
+        categories = http.request.env['helpdesk.ticket.category']. \
+            search([('active', '=', True)])
+        email = http.request.env.user.email
+        name = http.request.env.user.name
+        return http.request.render('helpdesk_mgmt.portal_create_ticket', {
+            'categories': categories, 'email': email, 'name': name})
+
+    @http.route('/submitted/ticket',
+                type="http", auth="user", website=True, csrf=True)
+    def submit_ticket(self, **kw):
```

Comment:
Please don't use `**kw`, as this is a security hole, allowing to overwrite any field. Limit the arguments to the ones you want to parse.

Reply:
Is it correct to do it with `request.params`? Although I have seen in the Odoo documentation that this is not usually used because `kw` provides you all the data.

Reply:
OK, seeing Odoo code, they use the same technique, although they call the variable as `**post`.